src/openrct2-ui/interface/Window.cpp

```diff
@@ -86,14 +86,20 @@ static bool WindowFitsOnScreen(const ScreenCoordsXY& loc, int32_t width, int32_t
     return WindowFitsBetweenOthers(loc, width, height);
 }
 
-static ScreenCoordsXY ClampWindowToScreen(const ScreenCoordsXY& pos, const int32_t screenWidth, const int32_t width)
+static ScreenCoordsXY ClampWindowToScreen(
+    const ScreenCoordsXY& pos, const int32_t screenWidth, const int32_t screenHeight, const int32_t width, const int32_t height)
 {
     auto screenPos = pos;
     if (screenPos.x < 0)
         screenPos.x = 0;
     if (screenPos.x + width > screenWidth)
         screenPos.x = screenWidth - width;
 
+    if (screenPos.y < 0)
+        screenPos.y = 0;
+    if (screenPos.y + height > screenHeight)
+        screenPos.y = screenHeight - height;
```

Comment:
If the window height (or width) is greater than the screen height then it's never going to fit properly. In which case we need to either center the window or prioritise one corner of the window. 

I think we want to prioritise showing the top of the window, and the right, so that the top and X is always visible.

Does that sound sensible?

Reply:
Yeah, I agree the title bar should always be visible. To achieve that, we may have to adjust for the top toolbar as well. That is, we should clamp to 28 (`TOP_TOOLBAR_HEIGHT` + 1), rather than clamping to 0.

Reply:
Clamping to 0 seems to still ensure the toolbar is visible from my testing

Reply:
The window will be rendered behind the toolbar, though, if there is any overlap. Particularly on lower (scaled) resolutions, this may pose problems.

Reply:
I have been an idiot and now understand, I was testing on the main menu. Will update